clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -282,6 +274,20 @@ private long appendEosRecord(long offset, long timestamp, ByteBuffer key, ByteBu
         return crc;
     }
 
+    private long appendOldRecord(long offset, long timestamp, ByteBuffer key, ByteBuffer value) throws IOException {
```

Comment:
Similar to my comment elsewhere, but I would prefer avoiding the use of the word `Old`. `V0` would be better since it is more explicit.